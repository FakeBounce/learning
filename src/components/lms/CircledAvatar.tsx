import { Box } from '@mui/material';
import { styled } from '@mui/system';
import { ReactNode } from 'react';

const StyledImageDisplayContainer = styled(Box)(({ theme }) => ({
  width: 128,
  height: 128,
  border: `1px dashed ${theme.palette.grey[200]}`,
  borderRadius: theme.shape.customBorderRadius.whole,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center'
}));

const StyledImageDisplayer = styled(Box)(({ theme }) => ({
  width: 112,
  height: 112,
  backgroundColor: theme.palette.grey[200],
  borderRadius: theme.shape.customBorderRadius.whole,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center'
}));

interface CircledAvatarProps {
  children: ReactNode;
  [key: string]: unknown;
}

export default function CircledAvatar({ children, ...other }: CircledAvatarProps) {
  return (
    <StyledImageDisplayContainer component="label" {...other}>
      <StyledImageDisplayer>{children}</StyledImageDisplayer>
    </StyledImageDisplayContainer>
  );
}
