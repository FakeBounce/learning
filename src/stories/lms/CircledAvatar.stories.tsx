import type { Meta, StoryObj } from '@storybook/react';
import CircledAvatar from '@src/components/lms/CircledAvatar';

const meta = {
  title: 'Example/CircledAvatar',
  component: CircledAvatar,
  parameters: {
    layout: 'centered',
    backgrounds: {
      default: 'dark'
    }
  },
  tags: ['autodocs'],
  args: {
    children: 'Avatar content'
  }
} as Meta<typeof CircledAvatar>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
