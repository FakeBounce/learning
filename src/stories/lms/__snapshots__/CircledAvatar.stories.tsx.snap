// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/CircledAvatar Default smoke-test 1`] = `
<label class="MuiBox-root css-fjsr0d">
  <div class="MuiBox-root css-jfdah8">
    Avatar content
  </div>
</label>
`;
